output/cloud/expv2/hdr.go

```diff
@@ -104,92 +94,72 @@ func (h *histogram) addToBucket(v float64) {
 		return
 	}
 
-	index := resolveBucketIndex(v)
+	ix := resolveBucketIndex(v)
+	if _, contains := h.Buckets[ix]; !contains {
+		h.trackBucket(ix)
+	}
 
-	// they grow the current Buckets slice if there isn't enough capacity.
-	//
-	// An example with growRight:
-	// With Buckets [4, 1] and index equals to 5
-	// then we expect a slice like [4,1,0,0,0,0]
-	// then the counter at 5th position will be incremented
-	// generating the final slice [4,1,0,0,0,1]
-	switch {
-	case len(h.Buckets) == 0:
-		h.init(index)
-	case index < h.FirstNotZeroBucket:
-		h.prependBuckets(index)
-	case index > h.LastNotZeroBucket:
-		h.appendBuckets(index)
-	default:
-		h.Buckets[index-h.FirstNotZeroBucket]++
+	if h.Buckets == nil {
+		h.Buckets = make(map[uint32]uint32)
```

Comment:
I think this should be done just when we make the histogram struct.

Reply:
I move it to a constructor